ui-src/lib/MoonfireAPI.js

```diff
@@ -0,0 +1,120 @@
+// vim: set et sw=2:
+//
+
+import $ from 'jquery';
+import URLBuilder from './support/URLBuilder';
+
+/**
+ * Class to insulate rest of app from details of Moonfire API.
+ *
+ * Can produce URLs for specifc operations, or a request that has been
+ * started and can have handlers attached.
+ */
+export default class MoonfireAPI {
+  /**
+   * Construct.
+   *
+   * The defaults correspond to a standard Moonfire installation on the
+   * same host that this code runs on.
+   *
+   * Requesting relative URLs effectively disregards the host and port options.
+   *
+   * @param  {String} options.host         Host where the API resides
+   * @param  {Number} options.port         Port on which the API resides
+   * @param  {[type]} options.relativeUrls True if we should produce relative urls
+   */
+  constructor({host = 'localhost', port = 8080, relativeUrls = true} = {}) {
+    const url = new URL('/api/', 'http://acme.com');
```

Comment:
Why is `http://acme.com` here? That seems to be a real domain that belongs to someone else, so let's find some other way to make this work.